File: backend/app/models/summary.py
"""Summary-related Pydantic models."""
from typing import Optional
from pydantic import BaseModel, Field


class SummaryResponse(BaseModel):
    """Response model for lecture summary."""
    lecture_id: str
    main_topics: list[str] = Field(..., description="Main topics covered (3-5 items)")
    key_definitions: list[dict[str, str]] = Field(
        default_factory=list,
        description="Key terms and their definitions"
    )
    important_facts: list[str] = Field(
        default_factory=list,
        description="Important facts, dates, numbers"
    )
    assignments: list[str] = Field(
        default_factory=list,
        description="Mentioned assignments or action items"
    )
    brief_summary: str = Field(..., description="Brief 2-3 sentence summary")
    language: Optional[str] = Field(None, description="Language of the summary")

File: frontend/lib/models/summary.dart
class KeyDefinition {
  final String term;
  final String definition;

  KeyDefinition({
    required this.term,
    required this.definition,
  });

  factory KeyDefinition.fromJson(Map<String, dynamic> json) {
    return KeyDefinition(
      term: json['term'] as String,
      definition: json['definition'] as String,
    );
  }
}

class Summary {
  final String lectureId;
  final List<String> mainTopics;
  final List<KeyDefinition> keyDefinitions;
  final List<String> importantFacts;
  final List<String> assignments;
  final String briefSummary;
  final String? language;

  Summary({
    required this.lectureId,
    required this.mainTopics,
    required this.keyDefinitions,
    required this.importantFacts,
    required this.assignments,
    required this.briefSummary,
    this.language,
  });

  factory Summary.fromJson(Map<String, dynamic> json) {
    return Summary(
      lectureId: json['lecture_id'] as String,
      mainTopics: (json['main_topics'] as List).cast<String>(),
      keyDefinitions: (json['key_definitions'] as List)
          .map((d) => KeyDefinition.fromJson(d as Map<String, dynamic>))
          .toList(),
      importantFacts: (json['important_facts'] as List).cast<String>(),
      assignments: (json['assignments'] as List).cast<String>(),
      briefSummary: json['brief_summary'] as String,
      language: json['language'] as String?,
    );
  }
}
